### PR TITLE
Feature/2862/divider

### LINKED ISSUE #2862: (SP: 4) Create Divider component aligned with design system
#### Description:
Implement a reusable `Divider` component that meets the design system standards. This component will be used throughout the application to maintain consistency in divider styles, behavior, and accessibility.

[Figma](https://www.figma.com/design/liJZDYhUnDP15Pi9bipDcv/Space2Study-(Students)?node-id=21376-87494&t=ATTIq5lxdqHxDboc-4)

#### Requirements:
- **Styles:**
  - Follow design system variables for colors, typography, padding, and other styles.
  
- **Props:**
  - `variant`: Defines type (linear, ellipse, middle).
  - `orientation`: Define orientation (vertical, horizontal)
  - `textAlign`: Define align for the text that will be provided by children's props
  - `type`: Define type (ellipse, linear)

Don't forget to cover component with tests and add to the storybook.

### PR DESCRIPTION
**Pull request by markgol777**
image
Added Divider and bunch of styles with it

Like:

- [X] Horizontal
- [X] Vertical
- [X] Ellipse 
- [X] A lot of styling options for the above

### CODE REVIEW

File: src/design-system/components/divider/Divider.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+.#{$prefix}divider {
+  width: 100%;
+  min-width: 200px;
+
+  &-ellipse {
+    border-radius: 50%;
+
+    &-sm {
+      background-color: #c8ccd0;
```
Comment (nebby2105) [inline src/design-system/components/divider/Divider.scss]: maybe add colors to variables file (? not sure)

File: src/design-system/components/divider/Divider.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    <Box
+      className={cn('s2s-divider', {
+        [`s2s-divider-${orientation}`]: orientation
+      })}
+    >
+      {type === 'linear' ? (
+        <Divider
+          className={'s2s-divider-line'}
+          orientation={orientation}
+          sx={{
```
Comment (nebby2105) [inline src/design-system/components/divider/Divider.tsx:44]: not sure if its needed but you can add a className here and move styles into your .scss file (optional)

File: src/design-system/components/divider/Divider.scss (hunk 1 above)
Comment (markgol777) [reply src/design-system/components/divider/Divider.scss]: Yeah that's a good suggestion

File: src/design-system/components/divider/Divider.tsx (hunk 2 above)
Comment (markgol777) [reply src/design-system/components/divider/Divider.tsx:44]: I'd rather keep it in sx for better code readability

**Review by nebby2105 — APPROVED** [review]

File: src/design-system/components/divider/Divider.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { cn } from '~/utils/cn'
+
+import '~scss-components/divider/Divider.scss'
+
+const variants = ['fullWidth', 'inset', 'middle'] as const
+const orientation = ['vertical', 'horizontal'] as const
+const thickness = ['sm', 'md', 'lg'] as const
+const textAlign = ['left', 'right', 'center'] as const
+const type = ['ellipse', 'linear'] as const
+const size = ['small', 'big'] as const
```
Comment (amoutens) [inline src/design-system/components/divider/Divider.tsx]: maybe better to replace big to large

File: src/design-system/components/divider/Divider.scss [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    &-lg {
+      background-color: $neutral-450;
+    }
+
+    &-small {
+      width: 8px;
+      height: 8px;
+    }
+
+    &-big {
```
Comment (amoutens) [inline src/design-system/components/divider/Divider.scss]: change to large

**Review by amoutens — CHANGES_REQUESTED** [review]

**Review by amoutens — APPROVED** [review]

File: src/design-system/components/divider/Divider.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+const DividerComponent: React.FC<DividerProps> = ({
+  variant,
+  orientation,
+  thickness,
+  size,
+  caption,
+  textAlign,
+  type
+}) => {
+  const width = thickness === 'lg' ? '#9199A1' : '#C8CCD0'
```
Comment (ShadowOfTheSpace) [inline src/design-system/components/divider/Divider.tsx]:
```suggestion
  const borderColor = thickness === 'lg' ? '#9199A1' : '#C8CCD0'
```
Rename it, please. It is not width

File: tests/unit/design-system/components/Divider/Divider.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  it('renders a linear divider with caption', () => {
+    render(
+      <DividerComponent
+        variant="fullWidth"
+        orientation="horizontal"
+        thickness="md"
+        textAlign="center"
+        caption="Test Caption"
+        type="linear"
+      />
```
Comment (ShadowOfTheSpace) [inline tests/unit/design-system/components/Divider/Divider.spec.jsx]: Replace double quotes here

File: tests/unit/design-system/components/Divider/Divider.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  it('renders a linear thick divider', () => {
+    render(
+      <DividerComponent
+        variant="fullWidth"
+        orientation="horizontal"
+        thickness="lg"
+        textAlign="center"
+        caption="Thick Divider"
+        type="linear"
+      />
```
Comment (ShadowOfTheSpace) [inline tests/unit/design-system/components/Divider/Divider.spec.jsx]: Same here

File: tests/unit/design-system/components/Divider/Divider.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    const { container } = render(
+      <DividerComponent
+        variant="inset"
+        orientation="horizontal"
+        thickness="sm"
+        type="ellipse"
+        size="small"
+        caption=""
+        textAlign="center"
+      />
```
Comment (ShadowOfTheSpace) [inline tests/unit/design-system/components/Divider/Divider.spec.jsx]: Add vscode extentions "Stylelint" and "Esilint". You will see all errors from our linter in vscode and also you will have an ability to auto fix all errors

File: tests/unit/design-system/components/Divider/Divider.spec.jsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+  it('renders a vertical linear divider', () => {
+    render(
+      <DividerComponent
+        variant="middle"
+        orientation="vertical"
+        thickness="md"
+        type="linear"
+        caption="Vertical Divider"
+        textAlign="center"
+      />
```
Comment (ShadowOfTheSpace) [inline tests/unit/design-system/components/Divider/Divider.spec.jsx]: Double quotes)

File: tests/unit/design-system/components/Divider/Divider.spec.jsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  it('applies custom thickness styling', () => {
+    render(
+      <DividerComponent
+        variant="middle"
+        orientation="horizontal"
+        thickness="lg"
+        type="linear"
+        caption="Styled Divider"
+        textAlign="left"
+      />
```
Comment (ShadowOfTheSpace) [inline tests/unit/design-system/components/Divider/Divider.spec.jsx]: Double quotes again)

**Review by ShadowOfTheSpace — CHANGES_REQUESTED** [review]

File: src/design-system/components/divider/Divider.tsx (hunk 5 above)
Comment (markgol777) [reply src/design-system/components/divider/Divider.tsx]: fair enough, my bad

**Review by ShadowOfTheSpace — APPROVED** [review]

**Pull request merged**